source/PlayerInfo.cpp

```diff
@@ -2455,6 +2461,40 @@ void PlayerInfo::StepMissions(UI *ui)
 				if(ship->IsDestroyed())
 					mission.Do(ShipEvent(nullptr, ship, ShipEvent::DESTROY), *this, ui);
 	
+	// Move all invalid missions to the end of the mission list.
+	missions.sort([] (const Mission &lhs, const Mission &rhs)
+	{
+		return lhs.IsValid();
+	});
+	auto mit = find_if(missions.begin(), missions.end(), [](const Mission &m) { return !m.IsValid(); });
+	if(mit != missions.end())
+	{
+		if(ui)
+		{
+			// Prompt the player about any paused missions, but if there are many
+			// do not name them all (since this would overflow the screen).
+			string message = "These active missions or jobs were deactivated due to a missing definition - perhaps you recently removed a plugin?\n";
+			auto it = mit;
+			int named = 0;
+			while(it != missions.end() && (++named < 10))
+			{
+				message += "\t\"" + (*it).Name() + "\"\n";
+				++it;
+			}
+			if(named >= 10 && it != missions.end())
+				message += " and " + to_string(distance(it, missions.end())) + " more.\n";
+			message += "They will be reactivated when the necessary plugin is reinstalled.";
+			ui->Push(new Dialog(message));
+		}
+		// Store any invalid missions on the inactive list.
+		inactiveMissions.splice(inactiveMissions.end(), missions, mit, missions.end());
```

Comment:
Moving the invalid missions to the inactiveMission-list before printing them on the UI feels a bit more natural.

Reply:
There isn't really a nice way to do that here, unfortunately, as the splice call doesn't move the `end()` iterator, nor does it return an iterator to the inserted range.

Reply:
I split this functionality up a bit, moving the invalid checks into `Load` and the UI notification into `Land`. The notification will only appear the first time the player enters the game after loading a save with affected missions.